app/controllers/search_controller.rb

```diff
@@ -4,19 +4,54 @@ class SearchController < ApplicationController
 
   ## return users of portal 'nucore'
   def user_search_results
-    raise NUCore::PermissionDenied unless session_user
-    @limit       = 25
-    @facility    = Facility.find(params[:facility_id]) if params[:facility_id]
-    @price_group = PriceGroup.find(params[:price_group_id]) if params[:price_group_id]
-    @account     = Account.find(params[:account_id]) if params[:account_id]
-    @product     = Product.find(params[:product_id]) if params[:product_id]
-    
-    term = generate_multipart_like_search_term(params[:search_term])
-    if params[:search_term].length > 0
-      conditions = ["(LOWER(first_name) LIKE ? OR LOWER(last_name) LIKE ? OR LOWER(username) LIKE ? OR LOWER(CONCAT(first_name, last_name)) LIKE ?)", term, term, term, term]
-      @users = User.find(:all, :conditions => conditions, :order => "last_name, first_name", :limit => @limit)
-      @count = User.count(:all, :conditions => conditions)
-    end
-    render :layout => false
+    raise NUCore::PermissionDenied if session_user.blank?
+    @limit = 25
+    load_facility
+    @price_group = PriceGroup.find(params[:price_group_id]) if params[:price_group_id].present?
+    @account = Account.find(params[:account_id]) if params[:account_id].present?
+    @product = Product.find(params[:product_id]) if params[:product_id].present?
+    load_users if params[:search_term].present?
+
+    render layout: false
+  end
+
+  private
+
+  def condition_sql
+    <<-SQL
+      (
+          LOWER(first_name) LIKE ?
+        OR
+          LOWER(last_name) LIKE ?
+        OR
+          LOWER(username) LIKE ?
+        OR
+          LOWER(CONCAT(first_name, last_name)) LIKE ?
+      )
+    SQL
+  end
+
+  def load_facility
+    @facility =
+      if params[:facility_id].present?
+        Facility.find(params[:facility_id])
+      else
+        all_facility
+      end
+  end
+
+  def query_conditions
+    @query_conditions ||=
+      [condition_sql, search_term, search_term, search_term, search_term]
+  end
+
+  def search_term
+    @search_term ||= generate_multipart_like_search_term(params[:search_term])
+  end
+
+  def load_users
+    user_relation = User.where(query_conditions)
```

Comment:
Would it be worth extracting a `UserFinder` or `UserQuery` service object? Get some of this out of the controller?